packages/runtime-core/src/helpers/resolveAssets.ts

```diff
@@ -106,13 +106,16 @@ function resolveAsset(
       resolve(instance[type] || (Component as ComponentOptions)[type], name) ||
       // global registration
       resolve(instance.appContext[type], name)
-
     if (!res && maybeSelfReference) {
       // fallback to implicit self-reference
       return Component
     }
 
-    if (__DEV__ && warnMissing && !res) {
+    if (
+      __DEV__ &&
+      warnMissing &&
+      ((!res && !isLateTag(name)) || (res && isLateTag(name)))
```

Comment:
Is the condition `res && isLateTag(name)` correct? When resolved and it's a late tag, why warn `Failed to resolve`?

Could you please also add some unit tests for it?

Reply:
The reason why no tests are added is that this PR has not been completed yet and it is invalid for `script setup`.
![image](https://github.com/vuejs/core/assets/32354856/84ecc8bd-4610-498b-958d-864460f85c06)
![image](https://github.com/vuejs/core/assets/32354856/ee8f5793-3e60-466c-b854-9249663a79a4)
It can be seen that when the component is introduced, the compiler does not compile and generate the calling code of resolveComponent, so no warning will be reported.